shared/src/types/storage.rs

```diff
@@ -157,6 +159,25 @@ impl core::fmt::Debug for BlockHash {
     }
 }
 
+/// The data from Tendermint header
+/// relevant for Anoma storage
+#[derive(Clone, Debug, BorshSerialize, BorshDeserialize)]
+pub struct Header {
+    /// Merkle root hash of block
+    pub hash: Hash,
+    /// Timestamp associated to block
+    pub time: DateTimeUtc,
+    /// Hash of the addresses of the next validator set
+    pub next_validators_hash: Hash,
+}
+
+impl Header {
+    /// The number of bytes when this header is encoded
+    pub fn encoded_len(&self) -> usize {
+        self.try_to_vec().map(|ser| ser.len()).unwrap_or(usize::MAX)
```

Comment:
How come we default to `usize::MAX` here? It feels like our `Header`s should always be serializable, ideally Borsh would have a `.to_vec()` that we could use that couldn't fail (apart from OOM?), otherwise I'd say to forward the error though it's less convenient to use.

Reply:
I could just unwrap here. The max size ensures that all checks down the line fail, but maybe a panic is better.

Reply:
yeah if the serialization shouldn't ever fail I'd say a comment on why we can directly `.unwrap()` would be less surprising than having in the code to return `usize::MAX`